### PR TITLE
for #392

### PR DESCRIPTION
**Pull request by ramvikrams**
added dict

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
can you add a test for this.  You can use the example from #392

**Comment by ramvikrams** [comment]
added the tests sir

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
CI is failing because you have to run `black` on the source.

File: tests/test_frame.py [hunk 1]
```diff
@@ -1915,3 +1915,8 @@ def f(s: pd.DataFrame) -> pd.Series:
         return s.mean()
 
     check(assert_type(resampler.apply(f), Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+    
+    
+def df_accepting_dicts_iterator() -> None:
```
[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_frame.py]: Done sir

File: tests/test_frame.py [hunk 2]
```diff
@@ -1915,3 +1915,8 @@ def f(s: pd.DataFrame) -> pd.Series:
         return s.mean()
 
     check(assert_type(resampler.apply(f), Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+    
+    
+def df_accepting_dicts_iterator() -> None: # GH 392
```
Comment (Dr-Irv) [inline tests/test_frame.py:1920]: Please put this comment on its own line (see elsewhere for examples of this)

You also have extra spaces on the two lines above.  Make sure you run `black` on your files before you commit.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_frame.py (hunk 2 above)
Comment (ramvikrams) [reply tests/test_frame.py]: Done and ran black

File: tests/test_frame.py [hunk 3]
```diff
@@ -1915,3 +1915,9 @@ def f(s: pd.DataFrame) -> pd.Series:
         return s.mean()
 
     check(assert_type(resampler.apply(f), Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+
+
+def df_accepting_dicts_iterator() -> None: 
+    # GH 392
+    data = [{"a": 1, "b": 2}, {"a": 3, "b": 5}]
+    p = pd.DataFrame(iter(data))
```
Comment (Dr-Irv) [inline tests/test_frame.py:1923]: One more thing.  Can you change this last line to:
```python
    check(assert_type(pd.DataFrame(iter(data)), pd.DataFrame), pd.DataFrame)
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_frame.py (hunk 3 above)
Comment (ramvikrams) [reply tests/test_frame.py:1923]: Sure

Comment (ramvikrams) [reply tests/test_frame.py]: Done

File: tests/test_frame.py [hunk 4]
```diff
@@ -1915,3 +1915,9 @@ def f(s: pd.DataFrame) -> pd.Series:
         return s.mean()
 
     check(assert_type(resampler.apply(f), Union[pd.Series, pd.DataFrame]), pd.DataFrame)
+
+
+def df_accepting_dicts_iterator() -> None:
```
Comment (Dr-Irv) [inline tests/test_frame.py:1920]: You have to make sure you run `black` before committing.  You have an extra space at the end of this line.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_frame.py (hunk 4 above)
Comment (ramvikrams) [reply tests/test_frame.py]: Done Sir

**Review by Dr-Irv — APPROVED** [review]
thanks @ramvikrams

**Comment by ramvikrams** [comment]
> thanks @ramvikrams 

Welcome sir you helped a lot

**Pull request merged**